ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -118,6 +118,13 @@ else if (desc.getPartitionSpec() != null) {
         Optional<String> partitionName = partitionMap.keySet().stream().findFirst();
         partitionName.ifPresent(compactionRequest::setPartitionname);
       }
+      if (table.getStorageHandler() != null && 
+          !table.getStorageHandler().canCompact(context.getConf(), table, compactionRequest.getPartitionname(), 
+              compactionRequest.getType())) {
+        LOG.info("Table=%s%s doesn't meet requirements for compaction", table.getTableName(), 
```

Comment:
Please add more description what requirement didn't met. 
e.g.: What was expected, what is the actual?

Reply:
I removed logging from this place and instead added it to `HiveIcebergStorageHandler#canCompact`.
This way it will log the exact reason why a table/partition doesn't meet compaction requirements.